### ISSUE TITLE
Typo/Inconistent naming in Pause helper functions

### ISSUE THREAD

**Issue by twitu** [NEARFoundation/near-contract-tools#70]
A method and macro attribute `standard` for the pause utility is named `paus` which looks inconsistent compared to other methods which are named with full `pause`.

https://github.com/NEARFoundation/near-contract-tools/blob/aad160bb15455eff3bb3b733d0e9da46bb01ceae/macros/src/pause.rs#L46-L49

https://github.com/NEARFoundation/near-contract-tools/blob/aad160bb15455eff3bb3b733d0e9da46bb01ceae/src/pause.rs#L25-L32

**Comment by encody** [comment]
The idea here is that different patterns use different prefixes (like `own` for owner). I kind of just made up `paus` on the spot for the pause-related functions & events. Open to discussion.

https://github.com/NEARFoundation/near-contract-tools/blob/aad160bb15455eff3bb3b733d0e9da46bb01ceae/src/owner.rs#L245-L265

**Comment by ryancwalsh** [comment]
I personally prefer verbosity. Abbreviations can be confusing and riskier (more subject to misinterpretation), especially across languages and cultures.

I like how Petar added this rule to our standard Eslint config that we're now using: https://github.com/sindresorhus/eslint-plugin-unicorn/blob/main/docs/rules/prevent-abbreviations.md

Do these function names even need prefixes? If yes, I wonder if it would make sense for the prefixes to be `owner_`, `pause_`, etc (or maybe something even longer, such as `owner_external_` or whatever could refer to a name of that section of code)?

**Comment by encody** [comment]
This is intended to be consistent with the pattern adopted by existing NEPs. For example:
* `nft_` for NEP-171
* `ft_` for NEP-141
* `mt_` for NEP-245

Although I also prefer verbosity, I think convention & consistency with the ecosystem is worth mentioning.

Also, at least to me, it is not obvious that `owner_` is a prefix (i.e. namespace), as opposed to simply a word at the beginning of a longer method name. That, of course, implies that `own_` is also not a good prefix for the owner pattern, which is a point I'm willing to concede.

**Comment by encody** [comment]
This is also relevant if we intend to [propose the components that expose external functions as NEPs](https://github.com/NEARFoundation/near-contract-tools/issues/32).

**Comment by ryancwalsh** [comment]
Yes, convention & consistency with the ecosystem is definitely worth considering.

Also, NEAR is new enough (and NCT may become important enough) that maybe this will help _define_ convention going forward.

I don't have strong opinions here. 

I'm curious what prefixes you're currently leaning towards @encody and also what @nearken , @sandoche , and @RobertNear think too.

**Comment by sandoche** [comment]
I agree with @encody, i'll avoid prefixes everywhere, except for standards such as ft, nft, mt

**Comment by twitu** [comment]
A couple of points to add to uniform naming convention, since I've been looking at more code.

Traits to expose external functions have the same keyword as prefix and suffix. Since prefix is needed to avoid namespace collision the suffix can be removed. It'll also result in shorter names without any loss of information. Along with the proposal for fully named prefixes the `OwnerExternal` would look something like this.

```
#[ext_contract(ext_owner)]
pub trait OwnerExternal {
    /// Returns the account ID of the current owner
    fn owner_get(&self) -> Option<AccountId>;

    /// Returns the account ID that the current owner has proposed take over ownership
    fn owner_get_proposed(&self) -> Option<AccountId>;
    ...
}
```

Currently the functions are `own_get_owner` and `own_get_proposed`.

**Comment by twitu** [comment]
@encody on your suggestion for naming convention. The separator could be a single underscore `_`, which would keep it consistent with snake case function naming convention. Double underscore would also give linting warnings https://github.com/rust-lang/rust/issues/19475.

If there's no specific reason for double underscores we could go with single. The other parts look good :+1:.

**Comment by encody** [comment]
The reason for the two underscores is to make it very clear that the prefix is a prefix. If there is some other, better character combination (that wouldn't then be easy to confuse with a non-standard method name), please let me know!

**Comment by twitu** [comment]
All the functions being exposed externally will be grouped together in the same trait. It will be visually clear that the same keyword is the prefix, a separate character for distinction may not be necessary. But I'm not opposed to it, if there's something that fits the function naming convention.

**Comment by encody** [comment]
Once the contract is compiled to WASM, the types (traits, etc.) are all erased, so if you query the WASM, only the exposed function names remain. Transactions calling the contract also don't have to reference (or even know about the existence of) a trait, so none of that information is present in a block explorer, for example.
